cloud/amazon/s3_website.py

```diff
@@ -0,0 +1,290 @@
+            changed = True
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        except ValueError as e:
+            module.fail_json(msg=str(e))
+    else:
+        try:
+            if (suffix is not None and website_config['IndexDocument']['Suffix'] != suffix) or \
+                    (error_key is not None and website_config['ErrorDocument']['Key'] != error_key) or \
+                    (redirect_all_requests is not None and website_config['RedirectAllRequestsTo'] != _create_redirect_dict(redirect_all_requests)):
+
+                try:
+                    bucket_website.put(WebsiteConfiguration=_create_website_configuration(suffix, error_key, redirect_all_requests))
+                    changed = True
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        except KeyError as e:
+            try:
+                bucket_website.put(WebsiteConfiguration=_create_website_configuration(suffix, error_key, redirect_all_requests))
+                changed = True
+            except (ClientError, ParamValidationError) as e:
+                module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        except ValueError as e:
+            module.fail_json(msg=str(e))
+
+        # Wait 5 secs before getting the website_config again to give it time to update
+        sleep(5)
+
+    website_config = client_connection.get_bucket_website(Bucket=bucket_name)
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(website_config))
+
+
+def disable_bucket_as_website(client_connection, module):
+
+    changed = False
+    bucket_name = module.params.get("name")
+
+    try:
+        client_connection.get_bucket_website(Bucket=bucket_name)
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchWebsiteConfiguration':
+            module.exit_json(changed=changed)
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    try:
+        client_connection.delete_bucket_website(Bucket=bucket_name)
+        changed = True
+    except ClientError as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    module.exit_json(changed=changed)
+
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(type='str', required=True),
+            state=dict(type='str', required=True, choices=['present', 'absent']),
+            suffix=dict(type='str', required=False, default='index.html'),
+            error_key=dict(type='str', required=False),
+            redirect_all_requests=dict(type='str', required=False)
+        )
+    )
+    
+    module = AnsibleModule(argument_spec=argument_spec,
+        mutually_exclusive = [
```

Comment:
indentation here is not right - `argument_spec` probably wants to be on the line below aligned with `mutually_exclusive` - https://www.python.org/dev/peps/pep-0008/#indentation